src/legacy/ledgerUtils.ts

```diff
@@ -492,3 +488,38 @@ export const signTxWithLedger = async (
     throw mapLedgerError(e)
   }
 }
+
+// not bumping react-native right now (couple to ledger)
+const BLUETOOTH_SCAN = 'android.permission.BLUETOOTH_SCAN'
+const BLUETOOTH_CONNECT = 'android.permission.BLUETOOTH_CONNECT'
+export const getLedgerPermissions = () => {
+  const permissions: Array<string> = [PermissionsAndroid.PERMISSIONS.ACCESS_FINE_LOCATION]
+  if (Platform.Version >= 31) {
+    permissions.push(BLUETOOTH_CONNECT)
+    permissions.push(BLUETOOTH_SCAN)
+  }
+  // hack
+  return permissions as unknown as Permission[]
+}
+
+export const requestLedgerPermissions = async () => {
+  if (Platform.OS !== 'android') return Promise.resolve()
+
+  const permissions = getLedgerPermissions()
+  const statuses = await PermissionsAndroid.requestMultiple(permissions)
+  const denied = Object.values(statuses).some((value) => value === 'denied')
+
+  return denied ? Promise.reject() : Promise.resolve()
```

Comment:
An error message in `Promise.reject` could be helpful to have in the future

Reply:
id also say its more idiomatic to throw errors in async functions and implicit return
```
...
  if (denied) throw ...
}
```